extensions/cl_extension_template.asciidoc

```diff
@@ -387,6 +387,29 @@ extensions, or the optional interactions are documented as Modifications
 to the other extension specs, then this section may be omitted.
 ****
 
+== Conformance tests
+
+****
+This section contains a description of the tests that an implementation
+must pass to be considered conformant with the specification. This section must
+be present for `khr` and `ext` extensions and is recommended for vendor
+extensions.
+
+If it is not possible to provide any tests for the specification, a
+justification must be provided here.
+
+Test descriptions must be numbered and their order maintained across
+successive revisions of the specification.
+
+****
+
+. The new *clShinyNewFunction* entrypoint must be called and succeed.
+. The value returned for +CL_FANCY_QUERY+ must be in the valid range.
+. Calling *clShinyNewFunction* twice with incompatible flags must fail
+ - Call *clShinyNewFunction* with +CL_NEW_FLAG+ set, it must succeed
+ - Call *clShinyNewFunction* again with +CL_NEW_INCOMPATIBLE_FLAG+ set, it must
+   fail
```

Comment:
I don't think we've traditionally done "negative testing" where we check for an error.  Do we want to recommend this?

Reply:
We do have some in the conformance tests (and probably should have more IMHO) but that's another discussion. I'll pick a less controversial example here.